src/qililab/digital/circuit_transpiler.py

```diff
@@ -54,11 +54,17 @@ def transpile_circuits(
         """Transpiles a list of ``qibo.models.Circuit`` to a list of pulse schedules.
 
         First makes a routing and placement of the circuit to the chip's physical qubits. And returns/logs the final layout of the qubits.
+        If passed, this is done, with the `placer`, `router` and `routing_iterations` params (if not, default ones are applied).
 
-        Then translates the circuit to a native gate circuit and applies virtual Z gates and phase corrections for CZ gates.
+        Then translates the circuit to the native gate circuit, of the chip (CZ, RZ, Drag, Wait and M (Measurement).
 
         And finally, it converts the native gate circuit to a pulse schedule using calibrated settings from the runcard.
 
+        If ``optimize=True`` (default behaviour), then it also does some circuit optimization:
+
+        - cancelling adjacent pairs of Hermitian gates (H, X, Y, Z, CNOT, CZ and SWAPs).
+        - applying virtual Z gates and phase corrections (adding up several pulses into a single one, commuting them with virtual Zs).
```

Comment:
Suggestion:

"""
Transpiles a list of ``qibo.models.Circuit`` objects into a list of pulse schedules.

The process involves the following steps:
1. Routing and Placement: Routes and places the circuit's logical qubits onto the chip's physical qubits. The final qubit layout is returned and/or logged. This step uses the `placer`, `router`, and `routing_iterations` parameters if provided; otherwise, default values are applied.
2. Native Gate Translation: Translates the circuit into the chip's native gate set (CZ, RZ, Drag, Wait, and M (Measurement)).
3. Pulse Schedule Conversion: Converts the native gate circuit into a pulse schedule using calibrated settings from the runcard.

If `optimize=True` (default behavior), the following optimizations are performed:
- Canceling adjacent pairs of Hermitian gates (H, X, Y, Z, CNOT, CZ, and SWAPs).
- Applying virtual Z gates and phase corrections by combining multiple pulses into a single one and commuting them with virtual Z gates.
"""

Reply:
Applied!